paneldb/server/views.py

```diff
@@ -16,18 +18,57 @@ def gene_panels():
 
 @app.route('/baitsets', methods=['GET', 'POST'])
 def baitsets():
-    LOG.info("IN BAITSETS")
-    document=None
+    """Handles the submission of baitsets"""
+
+    document = None
+    data = {}
     if request.method == 'POST': #add new baitset
+        # check if the post request has the file part
         if 'inputFile' not in request.files:
-            flash('No file part')
             return redirect(request.url)
 
         baits_file = request.files['inputFile']
+
+        # if user does not select a file, browser submit an empty part without filename
         if baits_file.filename == '':
             flash('No selected file')
             return redirect(request.url)
+
         else:
-            document=secure_filename(baits_file.filename)
+            # Parse the file to check that it's OK
+            # Extract fields to be passes to the DB
+            # Pass stuff to DB and get all updated baitsets in it.
+
+            document = secure_filename(baits_file.filename)
+
+
+                    # savefile(baits_file) How to use this function correctly?
+                    # Make the data dictionary work
+                   # baits_file.save(os.path.join(app.config['UPLOAD_FOLDER'], document))
+            savefile(baits_file ,document)
+
+            #get content from file:
+            test = read_file(os.path.join(app.config['UPLOAD_FOLDER'], document))
+
+
+
+            data = {
+                'document': document,
+                'upload_folder': app.config['UPLOAD_FOLDER'],
+                'bah' : test
+            }
+
+
+    return render_template("baitsets.html", **data)
+
+
+
+def allowed_file(file, filename):
```

Comment:
We don't need this function any more, because it's handled at the level of the HTML page

Reply:
ok